stats/otel.go

```diff
@@ -238,30 +244,37 @@ func (s *otelStats) getMeasurement(name, statType string, tags Tags) Measurement
 	}
 
 	// Clean up tags based on deployment type. No need to send workspace id tag for free tier customers.
+	newTags := make(Tags)
 	for k, v := range tags {
 		if strings.Trim(k, " ") == "" {
-			s.logger.Warnf("removing empty tag key with value %s for measurement %s", v, name)
-			delete(tags, k)
+			s.logger.Warnf("removing empty tag key with value %q for measurement %q", v, name)
+			continue
 		}
 		if _, ok := s.config.excludedTags[k]; ok {
-			delete(tags, k)
+			continue
 		}
```

Comment:
do we need this too?

Reply:
Same as above, better safe than sorry since we might not be the only ones using this lib and I'm not planning to check everywhere if removing it might break an exclusion :man_shrugging: